Methods - Generative Model (#9)
greenelab/text_mined_hetnet_manuscript: content/03.methods.md

## Training Models
### Generative Model
The generative model is a core part of this automatic annotation framework.

> This model incorporates multiple signals emitted by label functions and assigns the most appropriate training class to each candidate sentence.

Reviewer: ```suggestion
It integrates multiple signals emitted by label functions and assigns a training class to each candidate sentence.
```

This model assigns training classes by estimating the joint probability distribution of the latent true class ($Y$) and label function signals $(\Lambda$), $P(\Lambda, Y)$.
Assuming each label function is conditionally independent, the joint distribution is defined as follows:  

$$
P(\Lambda, Y) = \frac{\exp(\sum_{i=1}^{m} \theta^{T}F_{i}(\Lambda, y))}
{\sum_{\Lambda'}\sum_{y'} \exp(\sum_{i=1}^{m} \theta^{T}F_{i}(\Lambda', y'))}
$$  

where $m$ is the number of candidate sentences,$F$ is the vector of summary statistics and $\theta$ is a vector of weights for each summary statistic.
The summary statistics used by the generative model are as follows:  

$$F^{Lab}_{i,j}(\Lambda, Y) = \unicode{x1D7D9}\{\Lambda_{i,j} \neq 0\}$$
$$F^{Acc}_{i,j}(\Lambda, Y) = \unicode{x1D7D9}\{\Lambda_{i,j} = y_{i,j}\}$$   

> Where *Lab* stands for the label function propensity, which is the frequency of a label function emitting a signal.

Reviewer: ```suggestion
*Lab* is the label function's propensity (the frequency of a label function emitting a signal).
```

*Acc* is the individual label function's accuracy given the training class.

> This model finds the opimial weights ($\theta$) by minimizing the negative log likelihood:

Reviewer: ```suggestion
This model optimizes the weights ($\theta$) by minimizing the negative log likelihood:
```

$$\hat{\theta} = argmin_{\theta} -\sum_{\Lambda} log \sum_{Y}P(\Lambda, Y)$$

> Once optimized, the predictions from the generative model, $\hat{Y} = P(Y \mid \Lambda)$, are used as training classes for our dataset [@doi:10.1145/3209889.3209898; @doi:10.14778/3157794.3157797].

Reviewer: ```suggestion
In the framework we apply, predictions from the generative model, $\hat{Y} = P(Y \mid \Lambda)$ provide training classes for our dataset [@doi:10.1145/3209889.3209898; @doi:10.14778/3157794.3157797].
```
